test/body-parser.test.js

```diff
@@ -2,27 +2,41 @@ const assert = require('chai').assert;
 const bodyParser = require('../lib/body-parser');
 const EventEmitter = require('events');
 
-describe('tests body parser :', () => {
+describe('tests body parser middleware :', () => {
 
-	it('if no body, then calls "next()"', () => {
-		let bodyIn = '';
-		const parser = bodyParser (body => {
-			bodyIn = body;
-		});
+	it('reads request stream and returns body, then calls "next()"', done => {
+
+		const parser = bodyParser();
 		const req = new EventEmitter();
-		req.emit('data', 'test the body parser');
+		const next = () => {
+			// done();
+		};
+		let cat = {
+			"name": "mocha"
+		};
+		parser(req, null, next);
+		req.emit('data', JSON.stringify(cat));	
 		req.emit('end');
-		assert.deepEqual(bodyIn, 'test the body parser');
-
-
-		
+		assert.deepEqual(req.body, cat);
+		done();
 	});
 
-	it('if body, then returns body and calls "next()"', () => {
+	it('if no body, then calls "next()"', done => {
 
-	})
+		const parser = bodyParser();
+		const req = new EventEmitter();
+		const next = () => {
+			done();
+		};
+		let cat = {};
```

Comment:
Set `cat` to empty string. So, your condition in your `if` statement in `body-parser.js` passes the condition.